app/controllers/puppetclasses_controller.rb

```diff
@@ -85,4 +86,8 @@ def action_permission
         super
     end
   end
+
+  def fetch_associations
+    ActiveRecord::Associations::Preloader.new(@puppetclass.lookup_keys, [:lookup_values]).run
```

Comment:
The original code called .includes on the association, not on the original resource, causing the objects in the association to get reloaded.  Calling includes on the top level resource (Puppetclass), before setting attributes on it ought to work?